arm-advisor/2016-07-12-preview/swagger/advisor.json

```diff
@@ -0,0 +1,641 @@
+					"required": false,
+					"type": "string"
+				},
+				{
+					"name": "$top",
+					"in": "query",
+					"description": "The number of recommendations per page if using paged version of this API.",
+					"required": false,
+					"type": "integer"
+				},
+				{
+					"name": "$skipToken",
+					"in": "query",
+					"description": "The page continuation token if using paged version of this API.",
+					"required": false,
+					"type": "string"
+				}],
+				"responses": {
+					"200": {
+						"description": "OK",
+						"schema": {
+							"$ref": "#/definitions/CollectionResponse[ResourceRecommendationBase]"
+						}
+					}
+				},
+				"deprecated": false,
+				"x-ms-pageable": {
+					"nextLinkName": "nextLink"
+				},
+				"x-ms-examples": {
+					"ListRecommendations": {
+						"$ref": "../examples/ListRecommendations.json"
+					}
+				}
+			}
+		},
+		"/providers/Microsoft.Advisor/operations": {
+			"get": {
+				"tags": ["Operations"],
+				"description": "Lists all of the available Advisor REST API operations",
+				"operationId": "Operations_List",
+				"parameters": [{
+					"$ref": "#/parameters/apiVersionParameter"
+				}],
+				"responses": {
+					"200": {
+						"description": "OK",
+						"schema": {
+							"$ref": "#/definitions/CollectionResponse[OperationEntity]"
+						}
+					}
+				},
+				"deprecated": false,
+				"x-ms-pageable": {
+					"nextLinkName": "nextLink"
+				}
+			}
+		},
+		"/{resourceUri}/providers/Microsoft.Advisor/recommendations/{recommendationId}/suppressions/{name}": {
+			"get": {
+				"tags": ["Suppressions"],
+				"description": "Obtains detail of a suppression",
+				"operationId": "Suppressions_Get",
+				"parameters": [{
+					"name": "resourceUri",
+					"in": "path",
+					"description": "The fully qualified arm identifier of the resource for which the recommendation applies.",
+					"required": true,
+					"type": "string"
+				},
+				{
+					"name": "recommendationId",
+					"in": "path",
+					"description": "The recommendation Id",
+					"required": true,
+					"type": "string"
+				},
+				{
+					"name": "name",
+					"in": "path",
+					"description": "The suppression name.",
+					"required": true,
+					"type": "string"
+				},
+				{
+					"$ref": "#/parameters/apiVersionParameter"
+				}],
+				"responses": {
+					"200": {
+						"description": "OK",
+						"schema": {
+							"$ref": "#/definitions/Suppression"
+						}
+					}
+				},
+				"deprecated": false,
+				"x-ms-examples": {
+					"GetSuppressionDetail": {
+						"$ref": "../examples/GetSuppressionDetail.json"
+					}
+				}
+			},
+			"put": {
+				"tags": ["Suppressions"],
+				"description": "Provides the ability to snoozed/dismissed a recommendation. The snoozed/dismissed attribute of a recommendation is referred to as a suppression. Use this API to create or update the snoozed/dismissed status of a recommendation.",
+				"operationId": "Suppressions_Create",
```

Comment:
This operation can take care of create/update. This put call has reviewed and approved by ARM already.